### DISCUSSION TITLE
Is it possible to remove one row without re-render the entire table

### DISCUSSION THREAD

**Discussion by shinngtear** [TanStack/table#2448] [General] [answered]
For example, if I am passing users array (using useState [users, setUsers]) to the table, and on click of a button, modify users by setUsers(users.splice(1)). Currently I am seeing all rows and cells are getting re-rendered, is there anyway to avoid this?

**Comment by jdmcd** [comment]
@tannerlinsley I am facing this issue as well.

**Comment by gargroh** [comment, marked as answer]
Yeah...re-rendering of rows and cells will be done on change of data... it cannot be avoided as of now...

**Comment by jdmcd** [comment]
Maybe my issue is a bit different then... my problem is that when the parent component changes state (think opening a modal, etc) the entire table reloads as well even when there were no data changes

**Reply by gargroh** [reply]
try wrapping your table component in React.memo

**Reply by jdmcd** [reply]
@gargroh unfortunately the table component is being called from a component class - React.memo is only available in functional components right?

**Reply by gargroh** [reply]
it is intended to work with any component...

**Reply by gargroh** [reply]
or you can use pureComponent thing...
